src/crafting.cpp

```diff
@@ -24,6 +24,28 @@
 #include <math.h>    //sqrt
 #include <algorithm> //std::min
 
+enum usage {
+	use_from_map = 1,
+	use_from_player = 2,
+	use_from_both = 1 | 2,
+	use_from_none = 4,
+	cancel = 5
+};
+
+struct comp_selection {
+	usage use_from;
+	item_comp comp;
+
+	comp_selection() : use_from(usage::use_from_none), comp(item_comp("", 0)) { }
+};
+
+struct tool_selection {
+	usage use_from;
+	tool_comp comp;
+
+	tool_selection() : use_from(usage::use_from_none), comp(tool_comp("", 0)) { }
```

Comment:
yes, I wanted to do this, but I got some error like 'cannot assign to non-static member bla bla' it seems to work now though.